### ISSUE TITLE
False Negative: The IV generated can be false, please add necessary checks.

### ISSUE THREAD

**Issue by MHCreations** [kalessil/phpinspectionsea#1003]
Hello!
Great plugin, great tool! Many thanks! 

Issue Type: False Negative variable (value) detection. 

PhpStorm 2018.2.2
Build #PS-182.4129.45, built on August 22, 2018
JRE: 1.8.0_152-release-1248-b8 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0

PHP Inspections (EA EXtended) Version: 3.0.5.1

On the following sort of code block:

```
Do {
       $details['base_hash'] = \bin2hex(\openssl_random_pseudo_bytes(32, $innerStrong));
      }
while(!$innerStrong);
```

And likewise on 

```
$innerStrong = false;  
...
while(!$innerStrong) {
       $details['base_hash'] = \bin2hex(\openssl_random_pseudo_bytes(32, $innerStrong));
      }

```

This persists in telling me "The IV generated can be false, please add necessary checks." Whereas I DO have the checks in place _but in the preceeding lines!_ 

You should be notified this is occuring and maybe able to work magic in the plugin to detect variable values before the line of the IV Generator... 

Cheers

Screenshot: 

![false_neg](https://user-images.githubusercontent.com/17123773/44792539-a1545180-ab93-11e8-86e9-7acfb8449c9d.png)

------

Further: 
Testing shows that the bug is caused by the function being nested within another function (in this case `bin2hex(...)` .  See screenshot below which appears correctly: 

![false_neg2](https://user-images.githubusercontent.com/17123773/44792993-b67db000-ab94-11e8-9614-f4f558ed4a38.png)

See the only difference is that the result is nested within the outer function.

**Comment by kalessil** [comment]
The bug confirmed, thank you for reporting @MHCreations !

**Comment by kalessil** [comment]
Fixed!

**Comment by kalessil** [comment]
And updated samples. Sample 1:
```php
        do {
            $bytes = \openssl_random_pseudo_bytes(32, $innerStrong);
            // $bytes needs to be verified as well
        } while(!$bytes || !$innerStrong);
        $details['base_hash'] = \bin2hex($bytes);
```

**Comment by MHCreations** [comment]
Hello
With version 3.0.6 this issue still remains. Please see screenshot attached. 
![phpea306](https://user-images.githubusercontent.com/17123773/45951265-5858b700-bff2-11e8-94a1-929bc47dda1f.jpg)

**Comment by kalessil** [comment]
Re-opening in order to re-evaluate on the latest 3.0.6.1 version.

**Comment by MHCreations** [comment]
Sorry! I have found I needed to invalidate caches and restart PhpStorm itself (although the update to 3.0.6.1 happened days ago). This problem IS now fixed for me, thanks.

**Comment by MHCreations** [comment]
(Correction: invalidating caches does not fix anything.) 

Also attached is same issue with Do/while loop
![php-ea-306-2](https://user-images.githubusercontent.com/17123773/45951673-f436f280-bff3-11e8-8fc9-b2801e42e7ed.jpg)

Version 3.0.6.1

**Comment by kalessil** [comment]
:+1:

**Comment by kalessil** [comment]
@MHCreations: I've re-checked the case and the warning is correct, the function can return false and this scenario is not handled (here how it can be addressed: https://github.com/kalessil/phpinspectionsea/issues/1003#issuecomment-417277017).

The fix was related to handling `!$falsyVariable` constructs properly, sorry for misleading messages in the thread.
